drivers/interrupt_controller/intc_miwu.c

```diff
@@ -186,60 +183,60 @@ int npcx_miwu_interrupt_configure(const struct npcx_wui *wui,
 	const uint32_t base = DRV_CONFIG(miwu_devs[wui->table])->base;
 	uint8_t pmask = BIT(wui->bit);
 
-	if (mode == NPCX_MIWU_MODE_DISABLED) {
-		/* Clear MIWU enable bit */
-		NPCX_WKEN(base, wui->group) &= ~pmask;
+	/* If WK_EN is enabled, turn it off before configuring */
+	if (npcx_miwu_irq_get_state(wui)) {
```

Comment:
Is it worth making this conditional?  That is, why not just call `npcx_miwu_irq_disable(wui)`?

If it was enabled on input, should it be re-enabled before leaving?  Just in case somebody's relying on the previous behavior.

Reply:
> Is it worth making this conditional? That is, why not just call `npcx_miwu_irq_disable(wui)`?
> 
Although I have checked all cases meet the condition that WK_EN is zero before configuration, I still think it's worth adding it since the symptom is very hard to debug. (It might issue an interrupt unexpectedly.) And yes, I will call npcx_miwu_irq_disable(wui) directly instead of it.

> If it was enabled on input, should it be re-enabled before leaving? Just in case somebody's relying on the previous behavior.
Re-enable gpio or the other miwu input source before leaving this function might cause problems. For example,  the power rails of gpio's PU or PD might be gone (ie. floating) during booting. At this moment, ec has a chance to jump into ISR and calls some routines which are not initialized properly. (On Chromebook, gpio initializations have the highest priority than the other devices during booting.)

I think the timing to turn on miwu's interrupt which ISR relies on the other device is not this driver's responsibility. We should let the other device driver or upper layer decide when to enable it via npcx_miwu_enable(wui).

Reply:
Done.